src/api/chat_completions.rs

```diff
@@ -38,12 +38,19 @@ impl Instance {
             .await
             .map_err(|_| "Failed to fetch clue collection from API server")?;
 
-        // Deserialize the response
-        let clue_strings = response
+        let parsed_response = response
             .json::<ChatCompletionResponse>()
             .await
-            .map_err(|_| "Failed to parse clues from API server")?
-            .choices[0]
+            .map_err(|_| "Failed to parse clues from API server")?;
+
+        // Extract usage information from the parsed response
+        let usage = parsed_response.usage.clone();
```

Comment:
No need to clone here. You can simply transfer the ownership to `usage` since it's only used for return